### PR TITLE
feat: Adds `auto_indexing` attribute to `resource_serverless_instance`

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Adds `auto_indexing` attribute to `resource_serverless_instance`.

Also tests are refactored.

Link to any related issue(s): CLOUDP-216519

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: website/docs/r/serverless_instance.html.markdown [hunk 1]
```diff
@@ -46,9 +46,9 @@ Follow this example to [setup private connection to a serverless instance using
   Human-readable label that identifies the physical location of your MongoDB serverless instance. The region you choose can affect network latency for clients accessing your databases.
 * `continuous_backup_enabled` - (Optional) Flag that indicates whether the serverless instance uses [Serverless Continuous Backup](https://www.mongodb.com/docs/atlas/configure-serverless-backup). If this parameter is false or not used, the serverless instance uses [Basic Backup](https://www.mongodb.com/docs/atlas/configure-serverless-backup).  
 * `termination_protection_enabled` - Flag that indicates whether termination protection is enabled on the cluster. If set to true, MongoDB Cloud won't delete the cluster. If set to false, MongoDB Cloud will delete the cluster.
+* `auto_indexing` - (Optional) Flag that indicates whether the serverless instance uses [Serverless Auto Indexing](https://www.mongodb.com/docs/atlas/performance-advisor/auto-index-serverless/).
```
Comment (Zuhairahmed) [inline website/docs/r/serverless_instance.html.markdown]:
```suggestion
* `auto_indexing` - (Optional) `true` `false` Flag that indicates whether the serverless instance uses [Serverless Auto Indexing](https://www.mongodb.com/docs/atlas/performance-advisor/auto-index-serverless/).
```

File: website/docs/d/serverless_instances.html.markdown [hunk 2]
```diff
@@ -44,6 +44,7 @@ data "mongodbatlas_serverless_instances" "data_serverless" {
 * `state_name` - Stage of deployment of this serverless instance when the resource made its request.
 * `continuous_backup_enabled` - Flag that indicates whether the serverless instance uses Serverless Continuous Backup.
 * `termination_protection_enabled` - Flag that indicates whether termination protection is enabled on the cluster. If set to true, MongoDB Cloud won't delete the cluster. If set to false, MongoDB Cloud will delete the cluster.
+* `auto_indexing` - Flag that indicates whether the serverless instance uses [Serverless Auto Indexing](https://www.mongodb.com/docs/atlas/performance-advisor/auto-index-serverless/).
```
Comment (Zuhairahmed) [inline website/docs/d/serverless_instances.html.markdown:47]:
```suggestion
* `auto_indexing` - `true` `false` Flag that indicates whether the serverless instance uses [Serverless Auto Indexing](https://www.mongodb.com/docs/atlas/performance-advisor/auto-index-serverless/).
```

File: website/docs/d/serverless_instance.html.markdown [hunk 3]
```diff
@@ -53,6 +53,7 @@ Follow this example to [setup private connection to a serverless instance using
 * `state_name` - Stage of deployment of this serverless instance when the resource made its request.
 * `continuous_backup_enabled` - Flag that indicates whether the serverless instance uses Serverless Continuous Backup.
 * `termination_protection_enabled` - Flag that indicates whether termination protection is enabled on the cluster. If set to true, MongoDB Cloud won't delete the cluster. If set to false, MongoDB Cloud will delete the cluster.
+* `auto_indexing` - Flag that indicates whether the serverless instance uses [Serverless Auto Indexing](https://www.mongodb.com/docs/atlas/performance-advisor/auto-index-serverless/).
```
Comment (Zuhairahmed) [inline website/docs/d/serverless_instance.html.markdown:56]:
```suggestion
* `auto_indexing` - `true` false` Flag that indicates whether the serverless instance uses [Serverless Auto Indexing](https://www.mongodb.com/docs/atlas/performance-advisor/auto-index-serverless/).
```

**Review by Zuhairahmed — APPROVED** [review]
Minor tweak so users know what the input values are for the Flag, otherwise LGTM

File: website/docs/r/serverless_instance.html.markdown (hunk 1 above)
Comment (lantoli) [reply website/docs/r/serverless_instance.html.markdown]: as it's a flag it normally can have only true/false, maybe I can add default value to true

**Review by EspenAlbert — APPROVED** [review]
LGTM 👍

**Pull request merged**